### PR TITLE
Prompt user to leave rooms/subspaces in a space when leaving space

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/18071

image
image
image
image
image

### CODE REVIEW

File: res/css/views/dialogs/_LeaveSpaceDialog.scss [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    .mx_Dialog {
+        display: flex;
+        flex-direction: column;
+        padding: 24px 32px;
+    }
+}
+
+.mx_LeaveSpaceDialog {
+    width: 440px;
+    height: 500px;
```
Comment (novocaine) [inline res/css/views/dialogs/_LeaveSpaceDialog.scss]: this is potentially pretty big for a netbook, do we have any guidelines around supported screen sizes / dialog sizes?

Comment (t3chguy) [reply res/css/views/dialogs/_LeaveSpaceDialog.scss]: This is as specified by the design to maintain a non-jumpy experience when you toggle elements which cause the internal heights to change

We do not have any policy on supported screen sizes, the app does not function well on low resolution screens whatsoever, this doesn't change that. https://github.com/vector-im/element-web/issues/6488

**Review by dbkr — APPROVED** [review]
Also, "Space Child" would make an excellent DJ name

**Comment by niquewoodhouse** [comment]
Design review:

Having reviewed the flow in practice, now we don't have the transfer option it does feel a little strange to have the dropdown.  I'm not sure if it's the size of the dialog that makes me gloss over the dropdown, but @t3chguy you might have been right with radio buttons/checkboxes being better.  Will redesign and add a comment.  Thanks for suggesting it

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
Please could we proceed with changing the interaction from a drop down to 3 radio buttons, with the final one introducing more content. [Figma for reference](https://www.figma.com/file/Sx8vQd8cvrrjjIDrBz9uLw/Beta-Leave-a-space?node-id=153%3A28538).

https://user-images.githubusercontent.com/1587166/127324320-82c7a672-0cae-42a4-8696-ba104f2095ae.gif

**Comment by t3chguy** [comment]
image
image
image

Updated

**Review by niquewoodhouse — APPROVED** [review]

**Pull request merged**